tallycash/extension
# 🤖💡Tell Me You Get Me: Implement default on for analytics and notification

## Mandatory bad joke
<details>
  <summary>I get you ... the short story of Twilight</summary>
  
  ![CleanShot 2022-11-16 at 16 34 54](https://user-images.githubusercontent.com/7690112/202224220-eb88a29e-1a19-4d8b-a752-dd1f46c09ddd.png)
</details>

## What

Let's introduce the `default on` feature flag and behavior with notification for analytics. When this feature flag is flipped 
1) the analytics collection should start 
2) the user should see a notification 
3) the analytics menu item should become visible in the settings

The `SUPPORT_ANALYTICS` is a global feature flag that affects the sending of the analytics events on the lowest level. 
This is a user context flag that enables the feature. 

## How 

We have a settings flag in the preference service to detect when the `ENABLE_ANALYTICS_DEFAULT_ON` is turned on. This way, we can ship analytics but not collect any data before the users are notified.

The check happens [in main](https://github.com/tallycash/extension/blob/9b3e2b7ff895795639bbd05482b2b4710c7c0fa6/background/main.ts#L1422-L1433)

## Test

- start and load extension as is 
- no event should come through to `https://app.posthog.com/events`
- flip `SUPPORT_ANALYTICS=true` and restart the build
- no event should come through to `https://app.posthog.com/events`
- no notification should be visible
- flip `ENABLE_ANALYTICS_DEFAULT_ON=true` and set `USE_ANALYTICS_SOURCE=<myname>` and restart build 
- notification should be displayed
- Settings menu should appear
- events should start coming with a new uuid (there is a 60s delay)
- use the `X` on the notification > notification should disappear and stay that way
- in the settings turn off analytics
- events should stop flowing


## Testing Env

```
SUPPORT_ANALYTICS=true
ENABLE_ANALYTICS_DEFAULT_ON=true
```

## UI

![CleanShot 2022-11-16 at 16 17 50](https://user-images.githubusercontent.com/7690112/202220102-b3b7b8bf-7972-4a0b-b3d3-2d9dfebf0e4e.png)

Closes #2609

Latest build: [extension-builds-2636](https://github.com/tallycash/extension/suites/9358511602/artifacts/440549854) (as of Thu, 17 Nov 2022 10:54:08 GMT).
